### ISSUE TITLE
cli: generate account key if not specified

### ISSUE THREAD

**Issue by kanishkatn** [ChainSafe/gossamer#3255]
## Issue summary

If the account key is not specified in the cli flags, we need to generate one for the user.

**Comment by danforbes** [comment]
What is the current behavior?

**Comment by 1garo** [comment]
wanted to contribute to this issue, is it okay?

**Comment by 1garo** [comment]
> Of course @1garo ! I am off at the moment and can provide more context in a couple of days!

Sounds great, in the meantime I'm gonna take a look if I can have some progress.

**Comment by kanishkatn** [comment]
Here's where the account is being parsed: https://github.com/ChainSafe/gossamer/blob/development/cmd/gossamer/commands/utils.go#L322

With the current [default config setup](https://github.com/ChainSafe/gossamer/blob/development/chain/westend/defaults.go), we won't have a scenario where the account would be empty. We want to remove the default values for account and generate one it if it is empty.

Happy to answer any questions you have. If you're on ChainSafe discord, say hi in #gossamer for easy collab!

**Comment by 1garo** [comment]
I was able to generate the `default` keypair, I added a "default" case to `LoadKeystore`, and used `ed25519.GenerateKeypair()` and `sr25519.GenerateKeypair()` to generate the pairs (`ed25519` for grandpa).

It passes but later on broke on: 

```
2023-11-16T13:56:37-03:00 CRITICAL failed to run block production engine: cannot handle epoch: cannot initiate and get epoch handler: failed to initiate epoch: cannot get epoch data and start slot: cannot get latest epoch data: cannot get authority index: key not in BABE authority data	pkg=babe
```

**Comment by 1garo** [comment]
Seems like that by generating this new pair, their public key is not understand as a Authority, because it fails in the `getAuthorityIndex` function in the following code:

```go
	for i, auth := range Authorities {
		if bytes.Equal(pub.Encode(), auth.Key.Encode()) {
			return uint32(i), nil
		}
	}
```

**Comment by kanishkatn** [comment]
Hey @1garo , could you create a draft PR? We could move the conversation there!

**Comment by 1garo** [comment]
Of course, it's created: https://github.com/ChainSafe/gossamer/pull/3587

**Comment by kanishkatn** [comment]
I believe we don't have the need to generate an account if not specified since we're still in development and only need development accounts. @EclesioMeloJunior could you confirm?

Further, If we want to support the use of auto-generated or external accounts, we'll have to
- Add a new flag dev to enable the clear separation for the development runs
- Clear separation for the development config
- Store the auto generated account in disk
- Ability to load the account from disk

Thank you for your time and patience @1garo ! I believe this separation isn't a priority for us at the moment, but if you're interested in contributing to CLI, I can prolly create issues.

**Comment by 1garo** [comment]
> Thank you for your time and patience @1garo ! I believe this separation isn't a priority for us (core contributors) at the moment. If you're interested in contributing to CLI, I can create issues.

I would love to, and I feel that contributing to CLI could help me understand the gossamer better, so win-win.
